python/python/tests/test_async_client.py

```diff
@@ -7303,6 +7303,365 @@ async def test_lcs_idx(self, redis_client: GlideClient):
+        expected_keys = [f"{key}:{i}" for i in range(100)]
+        await redis_client.mset({k: "value" for k in expected_keys})
+        cursor = ClusterScanCursor()
+        keys: list[str] = []
+        while not cursor.is_finished():
+            result = await redis_client.scan(cursor)
+            cursor = cast(ClusterScanCursor, result[0])
+            result_keys = cast(list[str], result[1])
+            keys.extend(result_keys)
+
+        assert set(expected_keys) == set(keys)
+
+    @pytest.mark.parametrize("cluster_mode", [True])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_cluster_scan_with_object_type_and_pattern(
+        self, redis_client: GlideClusterClient
+    ):
+        key = get_random_string(10)
+        expected_keys = [f"key:{key}:{i}" for i in range(100)]
+        await redis_client.mset({k: "value" for k in expected_keys})
+        unexpected_type_keys = [f"{key}:{i}" for i in range(100, 200)]
+        for key in unexpected_type_keys:
+            await redis_client.sadd(key, ["value"])
+        unexpected_pattern_keys = [f"{i}" for i in range(200, 300)]
+        await redis_client.mset({k: "value" for k in unexpected_pattern_keys})
+        keys: list[str] = []
+        cursor = ClusterScanCursor()
+        while not cursor.is_finished():
+            result = await redis_client.scan(
+                cursor, match="key:*", type=ObjectType.STRING
+            )
+            cursor = cast(ClusterScanCursor, result[0])
+            result_keys = cast(list[str], result[1])
+            keys.extend(result_keys)
+
+        assert set(expected_keys) == set(keys)
+        assert not set(unexpected_type_keys).intersection(set(keys))
+        assert not set(unexpected_pattern_keys).intersection(set(keys))
+
+    @pytest.mark.parametrize("cluster_mode", [True])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_cluster_scan_with_count(self, redis_client: GlideClusterClient):
+        key = get_random_string(10)
+        expected_keys = [f"{key}:{i}" for i in range(100)]
+        await redis_client.mset({k: "value" for k in expected_keys})
+        cursor = ClusterScanCursor()
+        keys: list[str] = []
+        successful_compared_scans = 0
+        while not cursor.is_finished():
+            result_of_1 = await redis_client.scan(cursor, count=1)
+            cursor = cast(ClusterScanCursor, result_of_1[0])
+            result_keys_of_1 = cast(list[str], result_of_1[1])
+            keys.extend(result_keys_of_1)
+            if cursor.is_finished():
+                break
+            result_of_100 = await redis_client.scan(cursor, count=100)
+            cursor = cast(ClusterScanCursor, result_of_100[0])
+            result_keys_of_100 = cast(list[str], result_of_100[1])
+            keys.extend(result_keys_of_100)
+            if len(result_keys_of_100) > len(result_keys_of_1):
```

Comment:
Note that the result_of_keys_of_100 list could be small because there were few elements remaining in the data.

Reply:
True that why I checked few times and checked that at least one was bigger. The count is not deterministic so it's hard to test.